src/language/dafnyInstallation.ts

```diff
@@ -24,32 +28,65 @@ function ifNullOrEmpty(a: string | null, b: string): string {
   return a === null || a === '' ? b : a;
 }
 
-function getConfiguredVersion(): string {
-  const version = Configuration.get<string>(ConfigurationConstants.PreferredVersion);
-  return version === LanguageServerConstants.Latest
-    ? LanguageServerConstants.LatestVersion
-    : version;
+async function getConfiguredVersion(): Promise<string> {
+  const [ _, version ] = await getConfiguredTagAndVersion();
+  return version;
+}
+
+let getConfiguredTagAndVersionCache: [string, string];
+async function getConfiguredTagAndVersion(): Promise<[string, string]> {
+  if(getConfiguredTagAndVersionCache === undefined) {
+    const result = await getConfiguredTagAndVersionUncached();
+    if(getConfiguredTagAndVersionCache === undefined) {
+      getConfiguredTagAndVersionCache = result;
+    }
+  }
+  return getConfiguredTagAndVersionCache;
+}
+
+async function getConfiguredTagAndVersionUncached(): Promise<[string, string]> {
+  let version = Configuration.get<string>(ConfigurationConstants.PreferredVersion);
+  switch(version) {
+  case LanguageServerConstants.LatestStable:
+    version = LanguageServerConstants.LatestVersion;
+    break;
+  case LanguageServerConstants.LatestNightly: {
+    const result: any = await (await fetch('https://api.github.com/repos/dafny-lang/dafny/releases/tags/nightly')).json();
+    if(result.name !== undefined) {
+      const name: string = result.name;
+      const versionPrefix = 'Dafny ';
+      if(name.startsWith(versionPrefix)) {
+        const version = name.substring(versionPrefix.length);
+        return [ 'nightly', version ];
+      }
+    }
+    window.showWarningMessage('Failed to install latest nightly version of Dafny. Using latest stable version instead.\n'
```

Comment:
```suggestion
    window.showWarningMessage('Failed to install latest nightly version of Dafny. Using latest version instead.\n'
```
We don't have "latest stable" in our versions, only "latest". Alternatively, we could rename "latest" into "latest stable"?

Also, the ordering that we are providing (latest, 3.7, 3.6, ... 3.2, latest nightly) might be a bit misleading because latest nightly is even more later than latest. But I think I understand that we don't want users to use latest nightly by default. So I don't have a better advice than what you did.